openjdk/src/main/java/org/conscrypt/NativeCryptoJni.java

```
@@ -16,126 +16,97 @@
+                    classLoader(), results, platformLibName(), STATIC_LIB_NAME)) {
+            logResults(results);
+            throwBestError(results);
+        }
     }
 
     private NativeCryptoJni() {}
 
-    /**
-     * Normalizes the os.name value into the value used by the Maven os plugin
-     * (https://github.com/trustin/os-maven-plugin). This plugin is used to generate
-     * platform-specific
-     * classifiers for artifacts.
-     */
-    private static String normalizeOs(String value) {
-        value = normalize(value);
-        if (value.startsWith("aix")) {
-            return "aix";
-        }
-        if (value.startsWith("hpux")) {
-            return "hpux";
-        }
-        if (value.startsWith("os400")) {
-            // Avoid the names such as os4000
-            if (value.length() <= 5 || !Character.isDigit(value.charAt(5))) {
-                return "os400";
-            }
-        }
-        if (value.startsWith(LINUX)) {
-            return LINUX;
-        }
-        if (value.startsWith("macosx") || value.startsWith("osx")) {
-            return "osx";
-        }
-        if (value.startsWith("freebsd")) {
-            return "freebsd";
-        }
-        if (value.startsWith("openbsd")) {
-            return "openbsd";
-        }
-        if (value.startsWith("netbsd")) {
-            return "netbsd";
-        }
-        if (value.startsWith("solaris") || value.startsWith("sunos")) {
-            return "sunos";
-        }
-        if (value.startsWith("windows")) {
-            return "windows";
+    private static void logResults(List<LoadResult> results) {
+        for (LoadResult result : results) {
+            result.log();
         }
+    }
+
+    private static void throwBestError(List<LoadResult> results) {
+        Collections.sort(results, ErrorComparator.INSTANCE);
+        // Guaranteed to have at least one UnsatisfiedLinkError - use the best one.
+        throw (UnsatisfiedLinkError) results.get(0).error;
```

TIOLI: Swallowing the other errors seems potentially problematic, but most likely fine.  One suggestion would be to take a distinct set (based on exception.toString) of other errors, also excluding the selected best error.  Then use Throwable.html#addSuppressed to return them to callers.